File: apps/virtual-pet-adoption-js-admin/src/api/subscription/SubscriptionCreateInput.ts
import { PetWhereUniqueInput } from "../pet/PetWhereUniqueInput";
import { UserWhereUniqueInput } from "../user/UserWhereUniqueInput";

export type SubscriptionCreateInput = {
  endDate?: Date | null;
  pet?: PetWhereUniqueInput | null;
  startDate?: Date | null;
  status?: "Option1" | null;
  user?: UserWhereUniqueInput | null;
};

File: apps/virtual-pet-adoption-server/src/APIs/Subscription/Dtos/SubscriptionCreateInput.cs
using VirtualPetAdoption.Core.Enums;

namespace VirtualPetAdoption.APIs.Dtos;

public class SubscriptionCreateInput
{
    public DateTime CreatedAt { get; set; }

    public DateTime? EndDate { get; set; }

    public string? Id { get; set; }

    public Pet? Pet { get; set; }

    public DateTime? StartDate { get; set; }

    public StatusEnum? Status { get; set; }

    public DateTime UpdatedAt { get; set; }

    public User? User { get; set; }
}
